src/components/OfflineWithFeedback.js

```diff
@@ -0,0 +1,129 @@
+    errors: PropTypes.object,
+
+    /** A function to run when the X button next to the error is clicked */
+    onClose: PropTypes.func.isRequired,
+
+    /** The content that needs offline feedback */
+    children: PropTypes.node.isRequired,
+
+    /** Information about the network */
+    network: networkPropTypes.isRequired,
+
+    /** Additional styles to add after local styles. Applied to Pressable portion of button */
+    style: PropTypes.oneOfType([
+        PropTypes.arrayOf(PropTypes.object),
+        PropTypes.object,
+    ]),
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    pendingAction: null,
+    errors: null,
+    style: [],
+};
+
+/**
+ * This method applies the strikethrough to all the children passed recursively
+ * @param {Array} children
+ * @return {Array}
+ */
+function applyStrikeThrough(children) {
+    return React.Children.map(children, (child) => {
+        if (!React.isValidElement(child)) {
+            return child;
+        }
+        const props = {style: StyleUtils.combineStyles(child.props.style, styles.offlineFeedback.deleted)};
+        if (child.props.children) {
+            props.children = applyStrikeThrough(child.props.children);
+        }
+        return React.cloneElement(child, props);
+    });
+}
+
+const OfflineWithFeedback = (props) => {
+    const isOfflinePendingAction = props.network.isOffline && props.pendingAction;
+    const isUpdateOrDeleteError = props.errors && (props.pendingAction === 'delete' || props.pendingAction === 'update');
+    const isAddError = props.errors && props.pendingAction === 'add';
+    const needsOpacity = (isOfflinePendingAction && !isUpdateOrDeleteError) || isAddError;
+    const needsStrikeThrough = props.network.isOffline && props.pendingAction === 'delete';
+    const hideChildren = !props.network.isOffline && props.pendingAction === 'delete' && !props.errors;
+    let children = props.children;
+    const sortedErrors = _.chain(props.errors)
+        .keys()
+        .sortBy()
+        .map(key => props.errors[key])
+        .value();
+
+    // Apply strikethrough to children if needed, but skip it if we are not going to render them
+    if (needsStrikeThrough && !hideChildren) {
+        children = applyStrikeThrough(children);
+    }
+    return (
+        <View style={props.style}>
+            {!hideChildren && (
+                <View style={needsOpacity ? styles.offlineFeedback.pending : {}}>
+                    {children}
+                </View>
+            )}
+            {props.errors && (
+                <View style={styles.offlineFeedback.error}>
+                    <View style={styles.offlineFeedback.errorDot}>
+                        <Icon src={Expensicons.DotIndicator} fill={colors.red} height={16} width={16} />
+                    </View>
+                    <View style={styles.offlineFeedback.textContainer}>
+                        {_.map(sortedErrors, (error, i) => (
```

Comment:
It is not recommended to use indexes (`i`) as a `key`, from [react doc](https://reactjs.org/docs/lists-and-keys.html):

> We don’t recommend using indexes for keys if the order of items may change. This can negatively impact performance and may cause issues with component state. Check out Robin Pokorny’s article for an [in-depth explanation on the negative impacts of using an index as a key](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318). If you choose not to assign an explicit key to list items then React will default to using indexes as keys.

Not sure if it will have a real impact here... but an easy option to it would have been:

```js
const sortedErrorKeys = _.chain(props.errors)
        .keys()
        .sortBy()
       // .map(key => props.errors[key]) Don't map this to values!
        .value();

```
then, in your map here you would use the error's key as the key:

```
{_.map(sortedErrorKeys, (errorKey) => (
    <Text key={errorKey} style={styles.offlineFeedback.text}>{props.errors[errorKey]}</Text>
))}
```

Reply:
> We don’t recommend using indexes for keys if the order of items may change

The order of items will never change, so I think we are fine here, no?

Reply:
hmm the index for one of the errors may change if one of the errors is deleted